### PR TITLE
Refactoring dash components

### PR DESCRIPTION
**Pull request by mournfulCoroner**

### CODE REVIEW

**Comment by robot-charts** [comment]
**i18n-check**
<details>
<summary>The following components have not been translated</summary>
<p><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9962-dash-components-refactoting-1263/dash-error-view-1263/ru/">dash.error.view (ru)</a><br><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9962-dash-components-refactoting-1263/dash-error-view-1263/en/">dash.error.view (en)</a></p>
</details>

Wait for the reviewers to check your changes in Weblate and try running github action again.

**Comment by robot-charts** [comment]
**i18n-check**
<details>
<summary>The following components have not been translated</summary>
<p><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9962-dash-components-refactoting-1263/dash-error-view-1263/ru/">dash.error.view (ru)</a><br><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9962-dash-components-refactoting-1263/dash-error-view-1263/en/">dash.error.view (en)</a></p>
</details>

Wait for the reviewers to check your changes in Weblate and try running github action again.

File: src/ui/libs/DatalensChartkit/modules/data-provider/charts/index.ts [hunk 1]
```diff
@@ -710,7 +710,7 @@ class ChartsDataProvider implements DataProvider<ChartsProps, ChartsData, Cancel
             });
         }
 
-        const path = isEmbeddedChart() ? '/embeds/api/run' : '/api/run';
+        const path = isEmbeddedEntry() ? '/embeds/api/run' : '/api/run';
```
Comment (Marginy605) [inline src/ui/libs/DatalensChartkit/modules/data-provider/charts/index.ts:713]: Let's get this from DL.some_const

File: src/ui/units/dash/containers/Body/Body.tsx [hunk 2]
```diff
@@ -85,11 +84,28 @@ const b = block('dash-body');
...
 type DispatchProps = ResolveThunks<typeof mapDispatchToProps>;
 type OwnProps = {
+    noState?: boolean;
+    hideErrorDetails?: boolean;
+    onRetry: () => void;
+    globalParams: DashKitProps['globalParams'];
+    dashkitSettings: DashKitProps['settings'];
+} & (
+    | ({
+          noEdit?: boolean;
```
Comment (Marginy605) [inline src/ui/units/dash/containers/Body/Body.tsx:94]: editable or enableEdit or onlyView would be better

File: src/ui/units/dash/containers/Body/Body.tsx [hunk 3]
```diff
@@ -85,11 +84,28 @@ const b = block('dash-body');
 type StateProps = ReturnType<typeof mapStateToProps>;
 type DispatchProps = ResolveThunks<typeof mapDispatchToProps>;
 type OwnProps = {
+    noState?: boolean;
```
Comment (Marginy605) [inline src/ui/units/dash/containers/Body/Body.tsx:87]: enableState would be better

File: src/ui/units/dash/store/actions/base/actions.ts [hunk 4]
```diff
@@ -382,17 +350,7 @@ export const load = ({
...
-                        maxConcurrentRequests: data.settings.maxConcurrentRequests,
-                    });
-                }
-
-                if (data.settings.loadPriority) {
-                    isCallable(ChartKit.setDataProviderSettings)({
-                        loadPriority: data.settings.loadPriority,
-                    });
-                }
+                applyChartkitSettings({data});
```
Comment (Marginy605) [inline src/ui/units/dash/store/actions/base/actions.ts:353]: let's rename chartkit to dataproviderchart or something like this

File: src/ui/units/dash/store/actions/dashTyped.ts [hunk 5]
```diff
@@ -254,7 +254,7 @@ export type SetTabHashStateAction = {
     type: typeof SET_TAB_HASH_STATE;
     payload: {
         tabId: string;
-        entryId: string | null;
+        entryId?: string | null;
```
Comment (Marginy605) [inline src/ui/units/dash/store/actions/dashTyped.ts:257]: may be it's better use null in action?

**Review by Marginy605 — CHANGES_REQUESTED** [review]

**Comment by robot-charts** [comment]
**i18n-check**
<details>
<summary>The following components have not been translated</summary>
<p><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9962-dash-components-refactoting-1263/dash-error-view-1263/ru/">dash.error.view (ru)</a><br><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9962-dash-components-refactoting-1263/dash-error-view-1263/en/">dash.error.view (en)</a></p>
</details>

Wait for the reviewers to check your changes in Weblate and try running github action again.

**Comment by robot-charts** [comment]
**i18n-check**
<details>
<summary>The following components have not been translated</summary>
<p><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9962-dash-components-refactoting-1263/dash-error-view-1263/ru/">dash.error.view (ru)</a><br><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9962-dash-components-refactoting-1263/dash-error-view-1263/en/">dash.error.view (en)</a></p>
</details>

Wait for the reviewers to check your changes in Weblate and try running github action again.

File: src/ui/units/dash/hooks/useIframeFeatures.tsx [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+        function handleResize() {
+            sendEmbedDashHeight(wrapRef);
+        }
+
+        const throttledHandleResize = throttle(handleResize, 350);
+        const resizeObserver = new ResizeObserver(throttledHandleResize);
+        resizeObserver.observe(wrapRef.current);
+
+        return () => {
+            if (wrapRef.current) {
```
Comment (flops) [inline src/ui/units/dash/hooks/useIframeFeatures.tsx:74]: resizeObserver.disconnect()

**Comment by robot-charts** [comment]
**i18n-check**
<details>
<summary>The following components have not been translated</summary>
<p><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9962-dash-components-refactoting-1263/dash-error-view-1263/ru/">dash.error.view (ru)</a><br><a href="https://weblate.gravity-ui.com/projects/datalens/charts-9962-dash-components-refactoting-1263/dash-error-view-1263/en/">dash.error.view (en)</a></p>
</details>

Wait for the reviewers to check your changes in Weblate and try running github action again.

**Review by flops — APPROVED** [review]

**Review by Marginy605 — DISMISSED** [review]

**Review by flops — APPROVED** [review]

**Pull request merged**